lls/main.c

```diff
@@ -352,6 +358,87 @@ match_nd_router_sol_or_adv(struct rte_mbuf *pkt, struct gatekeeper_if *iface)
 	return 0;
 }
 
+static int
+submit_ping6(struct rte_mbuf **pkts, unsigned int num_pkts,
+	struct gatekeeper_if *iface)
+{
+	struct lls_ping6_req ping6_req = {
+		.num_pkts = num_pkts,
+		.iface = iface,
+	};
+	int ret;
+
+	RTE_VERIFY(num_pkts <= lls_conf.mailbox_max_pkt_sub);
+
+	rte_memcpy(ping6_req.pkts, pkts, sizeof(*ping6_req.pkts) * num_pkts);
```

Comment:
There's a serious bug here that affect this function (i.e. `submit_ping6()`), `submit_ping()` in the next patch,`submit_arp()`, and `submit_nd()`: `struct lls_ping6_req` doesn't have space for field `ping6_req.pkts`, so `rte_memcpy()` is invading memory.

The solution is to dynamically allocate the memory of `struct lls_ping6_req` on the stack using `alloca(3)`. You will need to create macros for the subexpressions of `ele_size` in `lls_stage1()` to reuse these macros to find the needed size of the allocation.

Please fix `submit_arp()`, and `submit_nd()` in another pull request. This way we can merge this new pull request before finishing the review of this pull request.